src/main/kotlin/com/sourcegraph/cody/chat/Chat.kt

```diff
@@ -50,52 +54,114 @@ class Chat {
         chat.updateLastMessage(chatMessage)
       }
     }
-    codyAgentServer
-        .thenAcceptAsync(
-            { server ->
-              try {
-                val recipesExecuteFuture =
-                    server.recipesExecute(
-                        ExecuteRecipeParams(recipeId, humanMessage.actualMessage()))
-                token.onCancellationRequested { recipesExecuteFuture.cancel(true) }
-                recipesExecuteFuture.handle { _, error ->
-                  if (error != null) {
-                    handleError(project, error, chat)
-                    null
-                  } else {
-                    RateLimitStateManager.invalidateForChat(project)
+    if (recipeId == "chat-question") {
+      codyAgentServer
+          .thenAcceptAsync(
+              { server ->
+                try {
+                  val reply =
+                      server.chatSubmitMessage(
+                          ChatSubmitMessageParams(
+                              chat.id!!,
+                              WebviewMessage(
+                                  command = "submit",
+                                  text = humanMessage.actualMessage(),
+                                  submitType = "user",
+                                  addEnhancedContext = true,
+                                  // TODO: allow to manually add files to the context via `@`
+                                  contextFiles = listOf())))
+                  token.onCancellationRequested { reply.cancel(true) }
+                  reply.handle { lastReply, error ->
+                    val rateLimitError =
+                        if (lastReply.type == "transcript" &&
```

Comment:
I think this can be done simpler and more idiomatic:

```kotlin
val error = lastReply.getError() // Return proper error class, e.g. PanelNotFoundError, all should inherit from common parent 
when (error) {
    is PanelNotFoundError -> ...
    is RateLimitError -> ...
    else -> ...
}
```